addons/mail/static/src/models/discuss/discuss.js

```diff
@@ -381,6 +399,12 @@ function factory(dependencies) {
             default: false,
             dependencies: ['isOpen'],
         }),
+        /**
+         * Determine whether this discuss should be focused at next render.
+         */
+        isDoFocus: attr({
+            default: false,
+        }),
```

Comment:
Not clear what it means for discuss to be focused, therefore what it's expected to do with `isDoFocus` set.
Generally `isDoFocus` should be very specific to the UI to make focus. That's why I'm not super fond with this approach to define `isDoFocus` on root components...

It's okay if user clicks on discuss or chat window and we want some special behaviour when focus is actually on these components (UI feedback and/or auto-focus cursor on text input, etc..), but if so it must feel natural. Here, if I'm not mistaken, this looks flow-dependent and specific to thread view part of discuss.

Reply:
As discussed the other day, focus should be a top level concern, because only one element on the page can be focused at a time.

Flux that "give" focus to `discuss` don't care and should not care what `discuss` is going to do with it. Maybe it will focus the thread viewer (which happens when displaying a normal thread), maybe it will focus the reply composer (inbox is open), or maybe there is no thread open at all, and it should focus the thread list to be able to navigate it with keyboard, etc.

"do focus" is a hint, and discuss component can do whatever it wants with it. When we set other hints we don't tell what component should do with them either.

Reply:
Note that "focusing thread viewer" is just as abstract as "focusing discuss", it so happens that thread viewer is typically focusing its composer, but that should not be assumed. As well as "focusing chat window" is not especially focusing composer because there be might a "new message" input or the chat window might even be folded instead.

Reply:
Maybe your suggestion would be more to be more precise with the hint name, instead of calling it "do focus". So we would have hint "clicked on expand", hint "clicked on discuss app", hint "return from home screen", etc.
Considering they all have the same result of just doing focus, it might be overkill to differentiate them now.

Reply:
`isDoFocus` was initially introduced with `mail.chat_window` as a way to cleanly manage focus from models. There is UI style on focused chat window, so it's clear what it means to do/have focus on chat window. Putting focus on composer is more an intentional side-effect, but I want to emphasize that it's an afterthought. It was also a mistake to let component deduce do focus on chat window => do focus on composer: should have been fully done in models.

_(BTW, I'm not sure it was a good idea to enforce this, because there may be some annoyances like in mobile it opens soft-keyboard when it shouldn't)_

What I don't like with `isDoFocus` here is that it's not clear what it means to have focus on Discuss. If user clicks anywhere on Discuss, Discuss should have the focus, therefore it should set `isDoFocus`... You see how wrong this is (especially given the name)?

At the very least, if we want to keep this way of managing focus in Discuss on this PR, the name of this field must be much more specific.

Also I disagree with approach that root components should manage focus themselves: it makes management of focus very cryptic like now with Discuss, and it encourages having inappropriate, complex and non-declarative code in components. `useRef` should be avoided, because it usually makes components much less declarative. Here it's definitely a bad use of refs.

The proper design is to have reference to item to focus in global object, like in `messaging`.

Reply:
Yes to have the global focus ref, but not in this PR, it's not possible without having a record for each view.

The one that is really an issue right now is the composer. Due to the "composer viewer/view" not existing, focus for it has to be on its closest "parent view", and this is not thread viewer (because discuss or chatter are directly instantiating composer too), so parent with "isDoFocus" must be discuss/chatter/chat window as done here.

Let's do a pass on this later, this PR is critical to be finished now.

Reply:
ok

Reply:
task-2327655